include/swift/IDE/SourceEntityWalker.h

```diff
@@ -41,6 +41,12 @@ namespace swift {
 /// An abstract class used to traverse the AST and provide source information.
 /// Visitation happens in source-order and compiler-generated semantic info,
 /// like implicit declarations, is ignored.
+/// If \c walkTo*Pre returns \c true, the children are visited and \c
+/// walkTo*Post is called after all children have been visited. If \c walkTo*Pre
+/// returns \c false, the corresponding \c walkTo*Post call will not be issued.
+/// Once the traversal is terminated, no more \c walk* calls are issued. Nodes
+/// that have already received a \c walkTo*Pre call will *not* receive a \c
+/// walkTo*Post call.
```

Comment:
Thanks for adding this comment! I really should have when I did my change in this class since all the true/falses/post being called/not really isn't obvious and took me a while to understand. You could also mention that if walkTo*Post returns false then the walk is cancelled.

Reply:
Good call. It’s now included in the comment.

Reply:
Thanks :)